src/functions/containers/FunctionEditPage.tsx

```diff
@@ -1,21 +1,107 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, lazy, Suspense, useContext, useEffect, useState} from 'react'
+import {useHistory, useParams} from 'react-router-dom'
+import {useSelector} from 'react-redux'
 
 // Components
-import {Page} from '@influxdata/clockface'
+import {
+  Page,
+  RemoteDataState,
+  SpinnerContainer,
+  TechnoSpinner,
+} from '@influxdata/clockface'
+import FunctionHeader from 'src/functions/components/FunctionHeader'
+import FunctionForm from 'src/functions/components/FunctionForm'
+import FunctionListProvider, {
+  FunctionListContext,
+} from 'src/functions/context/function.list'
+
+const FluxMonacoEditor = lazy(() =>
+  import('src/shared/components/FluxMonacoEditor')
+)
 
 // Utils
 import {pageTitleSuffixer} from 'src/shared/utils/pageTitles'
+import {getOrg} from 'src/organizations/selectors'
 
 const FunctionEditPage: FC = () => {
+  const history = useHistory()
+  const {id: functionID} = useParams<{id: string}>()
+  const {id: orgID} = useSelector(getOrg)
+
+  const [name, setName] = useState('New Function')
+  const [description, setDescription] = useState('')
+  const [params, setParams] = useState('')
+  const [script, setScript] = useState('')
+
+  const {functionsList} = useContext(FunctionListContext)
+  useEffect(() => {
+    const {name, script, description} = functionsList[functionID]
+    // TODO what if functionID no exist??
+    setName(name)
+    setScript(script)
+    setDescription(description)
+  }, [functionsList, functionID, setName, setScript, setDescription])
+
+  const {trigger} = useContext(FunctionListContext)
+
+  const updateFunction = () => {
+    // TODO route away if saved, stay if error
+  }
+
+  const cancelFunction = () => {
+    history.push(`/orgs/${orgID}/functions/`)
+    // TODO dont allow routing away if unsaved changes exist
+  }
+
+  const triggerFunction = () => {
+    trigger({script, params})
+  }
+
   return (
     <Page titleTag={pageTitleSuffixer(['Edit Function'])}>
-      <Page.Header fullWidth={false} testID="functions-edit-page--header">
-        <Page.Title title="Edit Function" />
-      </Page.Header>
-      <Page.Contents fullWidth={false} scrollable={true}></Page.Contents>
+      <FunctionHeader
+        name={name}
+        saveFunction={updateFunction}
+        cancelFunction={cancelFunction}
+      />
+      <Page.Contents fullWidth={true} scrollable={true}>
+        <div className="function-form">
+          <div className="function-form--options">
+            <FunctionForm
+              name={name}
```

Comment:
this interface feels weird. to pump all of these in individually, so that they can share state... idk what to do about it though, so lemme get back to it